### DISCUSSION TITLE
Recommended hardware (PCB/Kit) for Phazerville

### DISCUSSION THREAD

**Discussion by carlosedp** [djphazer/O_C-Phazerville#76] [General]
I'm kinda new into modular and O_C caught my attention with all it's functions... also I see Phazerville is one of the most featured firmware for it but I'm kinda confused on which hardware to run it.

Most PCBs I can find, support Teensy 3.2 which is practically non-existant and sold-out everywhere.

Which PCB/hardware is recommended to use Phazerville that supports Tennsy 4.0 or 4.1 which can be found?

Another question is if Phazerville supports some kind of MIDI output.

Thanks!

**Comment by xmacex** [comment]
There are second hand O_c modules going around, I would suggest you (too) to snatch one of those.

For MIDI out, yes you can use the (unsurprisingly named) [*MIDI out* app](https://github.com/Chysn/O_C-HemisphereSuite/wiki/MIDI-Out). Works well. The [wiki here in this repo](https://github.com/djphazer/O_C-Phazerville/wiki) lists and documents all the apps available. There is also this thing called [*Captain MIDI*](https://github.com/Chysn/O_C-HemisphereSuite/wiki/Captain-MIDI), which I have not used. There are a few YouTube videos which give a tour of what the Hemisphere/Benisphere/Phazerville family of O_c firmwares offers.

Also: welcome to modular, it's a long ride haha. O_c is a great place to start.

**Comment by djphazer** [comment]
To be clear, Teensy 4.0 will work as a direct replacement for Teensy 3.2 in all original O_C hardware shields. If you're building a new one DIY, either the 14HP or 8HP variant, just use a T4.0 ;)

**Comment by carlosedp** [comment]
@djphazer Oh nice, so I could get an uOC PCB and panel like https://pushermanproductions.com/product/mini-ornament-crime-pcb/ and https://pushermanproductions.com/product/antumbra-mini-ornament-crime-temps-utile-8hp-fr4-matte-black-panel/ or even a pre-soldered PCB like https://pushermanproductions.com/product/uoc-smd-pre-soldered-pcb-only/ from Pusherman, build it and use a Teensy 4.0 and put Phazerville into it with no issues?

@xmacex I understand I can have USB Midi in/out on it... but can I have MIDI on any of the jacks? Like to use it to sequence a module that have TRS Midi?

Thanks a lot guys!

**Comment by xmacex** [comment]
I better refrain from commenting on hardware side of things, ie. the the first topic – I just got a whole module second hand (which someone had built), it had Teensy 3.2 and is working super well. There are modules in circulation, and they'll come with a Teensy. Swap in a Teensy 4 if you feel like it (as djphazer said above).

Wups sorry for misunderstanding the MIDI question. USB midi only, no TRS MIDI on the jacks.
